### PR TITLE
[risk=low][no ticket] refactor some access-utils code

### PR DESCRIPTION
**Pull request by jmthibault79**
`hasRegisteredAccess()` -> `hasRegisteredTierAccess()`

add `hasTierAccess()`

a few other small changes

This will help with my work on https://precisionmedicineinitiative.atlassian.net/browse/RW-7520

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI and/or API server with `yarn test-local` or [yarn test-local-devup](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples)
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: ui/src/app/utils/access-utils.tsx [hunk 1]
```diff
@@ -295,7 +295,7 @@ export const computeDisplayDates = ({completionEpochMillis, expirationEpochMilli
 
 // return true if user is eligible for registered tier.
 // A user loses tier eligibility when they are removed from institution tier requirement
-export const eligibleForRegisteredForTier = (tierEligiblities: Array<UserTierEligibility>): boolean => {
+export const eligibleForRegisteredTier = (tierEligiblities: Array<UserTierEligibility>): boolean => {
```
Comment (jmthibault79) [inline ui/src/app/utils/access-utils.tsx:298]: typo

File: ui/src/app/components/side-nav.tsx [hunk 2]
```diff
@@ -285,22 +286,22 @@ export const SideNav = (props: SideNavProps) => {
         onToggleSideNav={() => onToggleSideNav()}
         href={'/workspaces'}
         active={workspacesActive()}
-        disabled={!hasRegisteredAccess(profile.accessTierShortNames)}
```
Comment (jmthibault79) [inline ui/src/app/components/side-nav.tsx:288]: changed the interface to Profile so the caller doesn't need to know what *part* of the Profile provides this information.

File: ui/src/app/utils/access-tiers.tsx [hunk 3]
```diff
@@ -13,13 +12,16 @@ export enum AccessTierDisplayNames {
...
  */
-export function hasRegisteredAccess(accessTierShortNames: Array<string>): boolean {
-  return fp.includes(AccessTierShortNames.Registered, accessTierShortNames);
+
+export function hasRegisteredTierAccess(profile: Profile): boolean {
+  return hasTierAccess(profile, AccessTierShortNames.Registered);
+}
+
+export function hasTierAccess(profile: Profile, shortName): boolean {
+  return profile.accessTierShortNames.includes(shortName);
```
Comment (jmthibault79) [inline ui/src/app/utils/access-tiers.tsx:24]: no need to use `fp` here when it's so simple

File: ui/src/app/utils/access-utils.tsx [hunk 4]
```diff
@@ -295,7 +295,7 @@ export const computeDisplayDates = ({completionEpochMillis, expirationEpochMilli
...
   const rtEligiblity = tierEligiblities.find(t => t.accessTierShortName === AccessTierShortNames.Registered)
-  return !!rtEligiblity && rtEligiblity.eligible
+  return rtEligiblity?.eligible
```
Comment (jmthibault79) [inline ui/src/app/utils/access-utils.tsx:300]: new Typescript features woo

**Review by yonghaoy — APPROVED** [review]

**Pull request merged**